ext/gd/gd.c

```diff
@@ -3254,6 +3254,63 @@ PHP_FUNCTION(imagecopyresized)
 }
 /* }}} */
 
+#define GET_TRUECOLOR_PIXEL(px, im, x, y) \
```

Comment:
I think we could use `gdImageRed()` and friends instead. Shouldn't make much of a performance difference.

https://github.com/php/php-src/blob/6fb79e269b926c1d0c4a116c6c5b00b377729547/ext/gd/libgd/gd.h#L771-L778